pkg/pillar/hypervisor/kvm.go

```diff
@@ -677,13 +770,51 @@ func (ctx kvmContext) Setup(status types.DomainStatus, config types.DomainConfig
 	return nil
 }
 
+func getFmlCustomResolution(status *types.DomainStatus, globalConfig *types.ConfigItemValueMap) (string, error) {
+	fmlResolutions := status.FmlCustomResolution
+	// if not set in the domain status, try to get it from the global config
+	if fmlResolutions == types.FmlResolutionUnset {
```

Comment:
@shjala, are you sure about this line? I the master branch I had to add a comparison with "" here. For some reason, it was a case...

Reply:
the status.FmlCustomResolution is always set from the domainmanager, either to a custom resolution that was read from the "cloud-config" or if nothing present in the "cloud-config" is set to Unset.

But now that you mentioned I think I should add a single line in case there is no "cloud-config" at all, but that shoulnt prevent us from testing, still can use the global config or "cloud-config".

Reply:
Yesterday, without this extra check for ", I got a situation when file resolution was set to ", and it resulted in the wrong filename and crash on app start. 

I HOPE IT WILL NOT BE THE CASE NOW, AFTER HOURS OF WAITING FOR THE HECKING WIN APP TO BE DOWNLOADED. Sorry =D

Reply:
I tested it on 9.4 multiple times, both cloud-config and global config where working fine.

Reply:
you can check for yourself `logread -f | grep FmlResolution`.

Reply:
My deployment process has yet to reach the point where this code is called...